File: fruitsIB2.c
int numOfUnplacedFruits(int* fruits, int fruitsSize, int* baskets, int basketsSize) {
    int unplaced = 0;

    for (int i = 0; i < fruitsSize; ++i) {
        int placed = 0;
        for (int j = 0; j < basketsSize; ++j) {
            if (baskets[j] >= 0 && baskets[j] >= fruits[i]) {
                baskets[j] = -1;  // Mark basket as used (negative value)
                placed = 1;
                break;
            }
        }
        if (!placed)
            ++unplaced;
    }

    return unplaced;
}

File: fruitsIB2.js
/**
 * @param {number[]} fruits
 * @param {number[]} baskets
 * @return {number}
 */
var numOfUnplacedFruits = function(fruits, baskets) {
    let unplaced = 0;

    for (let i = 0; i < fruits.length; i++) {
        let placed = false;
        for (let j = 0; j < baskets.length; j++) {
            if (baskets[j] >= fruits[i]) {
                baskets[j] = -1; // Mark basket as used
                placed = true;
                break;
            }
        }
        if (!placed) unplaced++;
    }

    return unplaced;
};
